great_expectations/expectations/metrics/util.py

```diff
@@ -666,73 +662,101 @@ def get_dbms_compatible_column_names(
     Args:
         column_names: Single string-valued column name or list of string-valued column names
         batch_columns_list: Properly typed column names (output of "table.columns" metric)
-        execution_engine: "ExecutionEngine" (here, of interest is whether or not "SqlAlchemyExecutionEngine" is used)
         error_message_template: String template to output error message if any column cannot be found in "Batch" object.
 
     Returns:
         Single property-typed column name object or list of property-typed column name objects (depending on input).
     """
-    verify_column_names_exist(
-        column_names=column_names,
-        batch_columns_list=batch_columns_list,
-        error_message_template=error_message_template,
+    normalized_typed_batch_columns_mappings: List[
+        Tuple[str, str | sqlalchemy.sql.quoted_name]
+    ] = (
+        verify_column_names_exist_and_get_normalized_typed_column_names_map(
+            column_names=column_names,
+            batch_columns_list=batch_columns_list,
+            error_message_template=error_message_template,
+        )
+        or []
     )
 
-    column_names_list: List[str]
-    is_list: bool
+    element: Tuple[str, str | sqlalchemy.sql.quoted_name]
+    typed_batch_column_names_list: List[str | sqlalchemy.sql.quoted_name] = [
+        element[1] for element in normalized_typed_batch_columns_mappings
+    ]
     if isinstance(column_names, list):
-        column_names_list = column_names
-        is_list = True
-    else:
-        column_names_list = [column_names]
-        is_list = False
-
-    typed_column_names_list: List[str | sqlalchemy.sql.quoted_name]
-    if isinstance(execution_engine, SqlAlchemyExecutionEngine):
-        column_name: str
-        batch_columns_dict: Dict[str, str | sqlalchemy.sql.quoted_name] = {
-            str(column_name): column_name for column_name in batch_columns_list
-        }
-        typed_column_names_list = [
-            batch_columns_dict[column_name] for column_name in column_names_list
-        ]
-    else:
-        typed_column_names_list = column_names_list
+        return typed_batch_column_names_list
 
-    if is_list:
-        return typed_column_names_list
-
-    return typed_column_names_list[0]
+    return typed_batch_column_names_list[0]
 
 
 def verify_column_names_exist(
     column_names: List[str] | str,
     batch_columns_list: List[str | sqlalchemy.sql.quoted_name],
     error_message_template: str = 'Error: The column "{column_name:s}" in BatchData does not exist.',
 ) -> None:
+    _ = verify_column_names_exist_and_get_normalized_typed_column_names_map(
+        column_names=column_names,
+        batch_columns_list=batch_columns_list,
+        error_message_template=error_message_template,
+        verify_only=True,
+    )
+
+
+def verify_column_names_exist_and_get_normalized_typed_column_names_map(
```

Comment:
Is this used externally outside of this file? If not, let's prefix with `_` to denote that it is private

Reply:
@cdkini Great idea -- done (committed).  Thanks.